src/entities/abTrail/ui/AbCanvas.tsx

```diff
@@ -252,63 +219,59 @@ const AbCanvas: FC<Props> = props => {
     return greenPointIndex;
   };
 
-  const onTouchStart = (touchInfo: TouchInfo) => {
+  const onTouchStart = (x: number, y: number, time: number) => {
     const isFinished = paths.length === currentIndexRef.current;
 
     if (currentPathRef.current || readonly || isFinished) {
+      resetCurrentPath();
       return;
     }
 
-    const point: Point = { x: touchInfo.x, y: touchInfo.y };
+    const point: Point = { x, y };
 
     if (isOverAny(point) && !isOverCurrent(point)) {
       onMessage(MessageType.IncorrectStartPoint);
+      resetCurrentPath();
       return;
     }
 
     if (!isOverCurrent(point)) {
+      resetCurrentPath();
```

Comment:
Why the new version contains 3 calls of the "resetCurrentPath" in this function, whereas the previous version doen't have any?

Reply:
@anq83  here, we leverage line creation to the `SketchCanvas` component. To keep the code as close as possible to the previous solution, it was decided to use `SketchCanvas` as `SkiaView`, which, in turn, automatically resets the drawing line on a touch lift. Hence, to implement such behavior, it is required to reset it manually

Reply:
ok